crates/kitsune_p2p/kitsune_p2p/tests/tests.rs

```diff
@@ -0,0 +1,134 @@
+// works and that networking works well enough for a request reply.
+#[cfg(feature = "tx5")]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_two_nodes_on_same_host_rpc_single() {
+    use std::sync::Arc;
+
+    use kitsune_p2p::actor::KitsuneP2pSender;
+
+    holochain_trace::test_run().unwrap();
+
+    let (bootstrap_addr, _bootstrap_handle) = start_bootstrap().await;
+    let (signal_url, _signal_srv_handle) = start_signal_srv().await;
+
+    let agent_store = Arc::new(parking_lot::RwLock::new(Vec::new()));
+    let host_api = Arc::new(TestHost::new(agent_store.clone()));
+    let mut harness_a = KitsuneTestHarness::try_new(host_api.clone())
+        .await
+        .expect("Failed to setup test harness")
+        .configure_tx5_network(signal_url)
+        .use_bootstrap_server(bootstrap_addr);
+
+    let mut harness_b = KitsuneTestHarness::try_new(host_api)
+        .await
+        .expect("Failed to setup test harness")
+        .configure_tx5_network(signal_url)
+        .use_bootstrap_server(bootstrap_addr);
+
+    let (sender_a, receiver_a) = harness_a
+        .spawn()
+        .await
+        .expect("should be able to spawn node");
+    let (sender_b, receiver_b) = harness_b
+        .spawn()
+        .await
+        .expect("should be able to spawn node");
+
+    let legacy_host_stub = TestLegacyHost::start(agent_store, vec![receiver_a, receiver_b]).await;
+
+    let space = Arc::new(fixt!(KitsuneSpace));
+    let agent_a = Arc::new(legacy_host_stub.create_agent().await);
+
+    sender_a
+        .join(space.clone(), agent_a, None, None)
+        .await
+        .unwrap();
+
+    let agent_b = Arc::new(legacy_host_stub.create_agent().await);
+
+    sender_b
+        .join(space.clone(), agent_b.clone(), None, None)
+        .await
+        .unwrap();
+
+    let resp = tokio::time::timeout(std::time::Duration::from_secs(10), async move {
+        loop {
+            match sender_a
+                .rpc_single(
+                    space.clone(),
+                    agent_b.clone(),
+                    "Hello from agent a".as_bytes().to_vec(),
+                    Some(std::time::Duration::from_secs(10).as_millis() as u64),
+                )
+                .await
+            {
+                Ok(resp) => {
+                    return resp;
+                }
+                Err(e) => {
+                    println!("Error sending rpc: {:?}", e);
+                    tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                }
+            }
+        }
+    })
+    .await
+    .unwrap();
+
+    // Assumes that the KitsuneP2pEvent::Call handler echoes the request
+    assert_eq!("Hello from agent a".as_bytes().to_vec(), resp);
+}
+
+#[cfg(feature = "tx5")]
+#[tokio::test(flavor = "multi_thread")]
+#[ignore = "This test deadlocks because the event receivers aren't consumed. This should not stall Kitsune"]
+async fn test_two_nodes_on_same_host_deadlock() {
```

Comment:
A curious assertion!

I suppose this problem will disappear if we get rid of the legacy API... but currently this channel facilitates function-like-calls into holochain, the responses of without which kitsune cannot proceed, so it seems appropriate to me that it deadlocks : )

Reply:
> I suppose this problem will disappear if we get rid of the legacy API

Actually, I think I was mistaken saying this. If holochain being slow is the problem, that won't change by switching from ghost_actor async calls to trait async calls. I'm not confident that the way this test is written just letting these messages backlog will result in the actual fix we want, but I also don't know that it won't : )

Reply:
Yes the test isn't useful, it would need to assert errors if we wanted to change the implementation. At the moment all you can observe is that the test will never finish